src/main/java/sirius/web/http/WebContext.java

```diff
@@ -1335,7 +1335,21 @@ public void setSessionCookie(String name, String value) {
      * @param value the contents of the cookie
      */
     public void setHTTPSessionCookie(String name, String value) {
-        setCookie(name, value, Long.MIN_VALUE, sessionCookieSameSite, sessionCookieSecurity);
+        setHTTPCookie(name, value, Long.MIN_VALUE);
+    }
+
+    /**
+     * Sets a http only cookie value to be sent back to the client.
+     * <p>
+     * The generated cookie will have a defined maximum lifetime. Also, this cookie
+     * will not be accessible by JavaScript and therefore slightly more secure.
+     *
+     * @param name          the cookie to create
+     * @param value         the contents of the cookie
+     * @param maxAgeSeconds contains the max age of this cookie in seconds
+     */
+    public void setHTTPCookie(String name, String value, long maxAgeSeconds) {
```

Comment:
Why is this an overload of `setCookie` but with a different name?

Reply:
It's not. It provides the possibility to define a custom max age for the cookie, whilst keeping the configured defaults regarding samesite and security.